processor/spanmetricsprocessor/processor.go

```diff
@@ -158,7 +158,7 @@ func newProcessor(logger *zap.Logger, config config.Processor, nextConsumer cons
 		return nil, err
 	}
 
-	internalRename, err := buildInternalRename(pConfig.Renames)
+	rename, err := buildRegexObjOnRename(pConfig.Renames)
```

Comment:
It would be better to be a method as the input and output are both the same struct. Being a method will also explicitly tell the user this struct is being mutated.